Simplenote/Classes/NSAttributedString+Styling.m

```diff
@@ -37,4 +39,51 @@ - (NSAttributedString *)attributedStringWithLeadingImage:(UIImage *)image lineHe
     return combinedString;
 }
 
+// Replaces checklist markdown syntax with SPTextAttachment images in an attributed string
++ (NSAttributedString *)attributedStringWithChecklistAttachments: (NSAttributedString *)sourceString withColor: (UIColor *)color {
+    if (!sourceString || sourceString.length == 0) {
+        return [[NSAttributedString alloc] initWithString:@""];
+    }
+    
+    NSError *error;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:CheckListRegExPattern options:NSRegularExpressionAnchorsMatchLines error:&error];
+    
+    NSString *noteString = sourceString.string;
+    NSArray *matches = [regex matchesInString:noteString options:0 range:[noteString rangeOfString:noteString]];
+    
+    if (matches.count == 0) {
+        return sourceString;
+    }
+    
+    NSMutableAttributedString *newString = [[NSMutableAttributedString alloc] initWithAttributedString:sourceString];
+    
+    int positionAdjustment = 0;
+    for (NSTextCheckingResult *match in matches) {
+        NSRange range = match.range;
+        if ([newString attribute:NSAttachmentAttributeName atIndex:match.range.location - positionAdjustment effectiveRange:&range]) {
+            // We've got one already!
+            continue;
+        }
+        
+        NSString *markdownTag = [noteString substringWithRange:match.range];
+        BOOL isChecked = [markdownTag containsString:@"x"];
+        
+        SPTextAttachment *attachment = [[SPTextAttachment alloc] init];
+        UIImage *image = [[UIImage imageNamed: isChecked ? @"icon_task_checked" : @"icon_task_unchecked"]
+                          imageWithOverlayColor:color];
+        [attachment setIsChecked: isChecked];
```

Comment:
Since this snippet is also elsewhere... maybe it should live inside SPTextAttachment?

(Whenever you set the isChecked, the internal image is toggled?)